backend/src/models/Scan.ts

```diff
@@ -0,0 +1,110 @@
+import { model, ObjectId, Schema } from 'mongoose'
+import MongooseDelete, { SoftDeleteDocument } from 'mongoose-delete'
+
+import { ScanState, ScanStateKeys } from '../connectors/fileScanning/Base.js'
+
+// This interface stores information about the properties on the base object.
+// It should be used for plain object representations, e.g. for sending to the
+// client.
+export interface ScanInterface {
+  _id: ObjectId
+
+  // file or image
+  artefactType: ArtefactTypeKeys
+  // for files
+  fileId?: string
+  // for images
+  // map this to registry GET /v2/<name>/manifests/<reference>
+  repositoryName?: string
+  imageDigest?: string
+
+  toolName: string
+  scannerVersion?: string
+  state: ScanStateKeys
+  isInfected?: boolean
+  viruses?: string[]
+  lastRunAt: Date
+
+  createdAt: Date
+  updatedAt: Date
+}
+
+export const ArtefactType = {
```

Comment:
We often replace `type` with `kind`, just to avoid confusing it with TS types. So I would replace this with `ArtefactKind`